Xlib/support/unix_connect.py

```diff
@@ -127,11 +112,37 @@ def get_socket(dname, protocol, host, dno):
         raise error.DisplayConnectionError(dname, str(val))
 
     # Make sure that the connection isn't inherited in child processes.
-    fcntl.fcntl(s.fileno(), F_SETFD, FD_CLOEXEC)
+    _ensure_not_inheritable(s)
 
     return s
 
 
+def _ensure_not_inheritable(sock):
+    # According to PEP446, in Python 3.4 and above,
+    # it is not inherited in child processes by default.
+    # However, just in case, we explicitly make it non-inheritable.
+    # Also, we don't use the code like the following,
+    # because there would be no possibility of backporting to past versions.
+    #   if sys.version_info.major == 3 and sys.version_info.minor >= 4:
+    #       sock.set_inheritable(False)
+    #       return
+    # We just check if the socket has `set_inheritable`.
+    if hasattr(sock, 'set_inheritable'):
+        sock.set_inheritable(False)
+        return
+
+    # On Windows,
+    # Python doesn't support fcntl module because Windows doesn't have fcntl API.
+    # At least by not importing fcntl, we will be able to import python-xlib on Windows.
+    if platform.system() == 'Windows':
+        # so.. unfortunately, for Python 3.3 and below, on Windows,
+        # we can't make sure that the connection isn't inherited in child processes for now.
+        return
+
+    import fcntl
+    fcntl.fcntl(sock.fileno(), fcntl.F_SETFD, fcntl.FD_CLOEXEC)
```

Comment:
Does it make sense to keep condition `if hasattr(fcntl, 'F_SETFD'):` as in the old code?

Reply:
The code below is the old code.
https://github.com/python-xlib/python-xlib/blob/376703ebb0069cf01727e22a679590fc14ac0152/Xlib/support/unix_connect.py#L27-L40

tbh, I'm not sure the condition should be kept or not.. 
but probably no, because there seems to be not `FCNTL` module in python 3.x, so.. even if ` hasattr(fcntl, 'F_SETFD')` is false, we might not be able to use the fallback code like use `FCNTL`'s constants. Also, I guess `socket.socket.set_inheritable` would exist for most env in Python 3.4 or above, so basically the new code itself uses `fcntl` module is fallback/alternative code just in case.

Reply:
oh.. sorry, I now noticed [this](https://github.com/python-xlib/python-xlib/blob/master/setup.cfg#L22-L24)
https://github.com/python-xlib/python-xlib/blob/master/setup.cfg#L22-L24
```
	Programming Language :: Python :: 2.7
	Programming Language :: Python :: 3
	Programming Language :: Python :: 3.3
```
python-xlib supports Python 2.7, so if Python 2.7 has `FNCTL` module, it might be better to use the condition and `FNCTL` of the old code..

Reply:
I couldn't import `FCNTL` in Python 2.7.18 on WSL2(ubuntu) at least.

Reply:
I think we can easily drop Py3.3, 3.4 and 3.5 support.

Reply:
`F_SETFD` and `FD_CLOEXEC` are known constants anyway. The name are just there to avoid magic numbers. It can easily be simplified to:
```py
# At the top of the file to avoid try-except overhead everytime the method is ran
try:
    import fcntl

    FD_CLOEXEC = getattr(fcntl, 'FD_CLOEXEC', 1)
    F_SETFD = getattr(fcntl, 'F_SETFD', 2)
except ImportError:
    FD_CLOEXEC = 1
    F_SETFD = 2
```
Or even forgo the entire try block at that point.

Reply:
> I think we can easily drop Py3.3, 3.4 and 3.5 support.

On a tangential note (the reason why I came across this import issue as well), dropping support for python 3.4 and below would allow easier typing support without having to rely on type-comments.